src/server.h

```diff
@@ -1484,12 +1518,9 @@ struct redisServer {
     long long second_replid_offset; /* Accept offsets up to this for replid2. */
     int slaveseldb;                 /* Last SELECTed DB in replication output */
     int repl_ping_slave_period;     /* Master pings the slave every N seconds */
-    char *repl_backlog;             /* Replication backlog for partial syncs */
+    struct replBacklogRefReplBuf *repl_backlog; /* Replication backlog for partial syncs */
```

Comment:
This way breaks too many unnecessary places.
@oranagra  WDYT？

Reply:
i see 11 usages of one, and 18 usages of the other (which is not that many),
but more importantly, it looks like nearly any place that uses or sets them is also dealing with `server.repl_backlog_idx` (which we did change, so we'll be changing a block that we had to change anyway).

i think the only places that only care about these two and not the others are `masterTryPartialResynchronization` and `genRedisInfoString`, i.e. they care about the range, but not about the data.
so i'm willing to consider this suggestion as if it only affects two functions, so if we feel it makes the code cleaner, let's go for it.

i.e. the two things to avoid in such renames are blame log damage causing untraceable history (we modify many of these blocks anyway), and merge conflicts (with the two exceptions above, any fix touching these areas will have hard time being backported or merged to some odd fork anyway)

Reply:
it seems that `repl_backlog_size` shouldn't be in it, since it is a configuration.